app/database_etl/test_evaluation_handler/test_eval_link_handler.py

```diff
@@ -0,0 +1,100 @@
+import pandas as pd
+
+TEST_EVAL_CSV_LOCATION = 'test_evaluation_handler/test_evaluations.csv'
+
+
+def get_isotype_list(isotype_str):
+    if pd.isna(isotype_str): return []
+    if isotype_str == "Total Ig": return ["IgM", "IgG", "IgA"]
+    isotypes = isotype_str.split('; ')
+    return set(isotypes)
+
+
+test_eval_df = pd.read_csv(TEST_EVAL_CSV_LOCATION)
+test_eval_df['Target'] = test_eval_df.apply(lambda x: get_isotype_list(x['Target']), axis=1)
+manufacturer_mapping = {
+    'Vircell Microbiologists': 'Vircell S.L.',
+    'SNIBE – Shenzhen New Industries Biomedical Engineering Co.': 'Snibe Co., Ltd',
+    'Zhejiang Orient Gene (Biotech Co LTD China)': 'Orient Gene Biotech',
+    'VivaCheck Biotech': 'VivaChek',
+    'Roche Diagnostics': 'Roche'
+}
+
+
+def get_eval_UID_and_DOI(row):
+    if not row['test_manufacturer'] or not row['isotypes']: return [None]*2
+    if row['ind_eval_link'] or row['test_linked_uid']: return [row['test_linked_uid'], row['ind_eval_link']]
+    test_name, manufacturers, target_isotypes = row['test_name'], row['test_manufacturer'], row['isotypes']
+
+    # CRITERIA A – test name, manufacturer, and target isotype
+    # Get all rows matching test name
+    matches = test_eval_df[test_eval_df['TestName'] == test_name]
+    if matches.empty: return [None]*2
+
+    if manufacturers[0] == 'Zhejiang Orient Gene (Biotech Co LTD':
+        manufacturers = ['Zhejiang Orient Gene (Biotech Co LTD China)']
+
+    temp = pd.DataFrame()
+    # Get all rows matching any name in manufacturer list
+    for manufacturer in manufacturers:
+        if manufacturer in manufacturer_mapping: manufacturer = manufacturer_mapping[manufacturer]
+        manu = matches[matches['Manufacturer'].str.contains(f"(?i){manufacturer}")]
+        temp = temp.append(manu)
+
+    matches = temp
+    # Check that targets are exact match
+    for i, ev in matches.iterrows():
```

Comment:
`iterrows` and `drop` workflow never usually ideal in pandas. You could do this one better with a vectorized approach. Something like:

matches = matches[matches.apply(lambda match: set(match['Target']) == target_isotypes]